src/simplify_reshapes.cpp

```diff
@@ -1065,16 +1065,14 @@ struct find_reshape_reshape_dot
 // const folding simplifications
 struct find_mul_add_shape_op_dot
 {
-    std::string op_name;
-
     auto matcher() const
     {
-        auto const_mul_add = match::name("mul", "add")(
+        const std::unordered_set<std::string> shape_ops = {"transpose", "convert"};
```

Comment:
Could you add a comment as to what logic this is doing?  I'm trying to work it out just by reading the code, but functions such as `match::skip_output` are just so cryptic that I can't tell if this is a functional change or merely style.